### PR TITLE
Minimal score tracking

### PR DESCRIPTION
**Pull request by sveinng**
# What?

Minimal score tracking
Option to reset game - new players
# Why?

Keep track of score.
# Reviewer checklist

[ ] - The change introduces no untested code
[ ] - The change passes all existing tests
[ ] - The change introduces no typos or formatting errors

### CODE REVIEW

**Review by eirikura — APPROVED** [review]

File: src/main/java/is/ru/tictactoe/webui/WebUI.java [hunk 1]
```diff
@@ -166,8 +180,26 @@ public static Object resetGameHandler(Request request, Response response) {
 
         return null;
     }
+    
+    
+    public static Object newGameHandler(Request request, Response response) {
+        request.session().removeAttribute("game");
+        request.session().removeAttribute(GAMETYPE);
```
Comment (solvip) [inline src/main/java/is/ru/tictactoe/webui/WebUI.java]: This mixing of string constant and constant variables feels iffy

File: src/main/java/is/ru/tictactoe/webui/WebUI.java [hunk 2]
```diff
@@ -21,6 +21,7 @@
     private static final String GAMETYPE = "unknown";
     private static Player player1;
     private static Player player2;
+    private static int numberOfGames = 0;
```
Comment (solvip) [inline src/main/java/is/ru/tictactoe/webui/WebUI.java]: This must have slipped review last time.

Use of static variables here will probably not do what you want, as concurrent clients will update them independently.  There is only _one_ instance of them for the entire web server.

We want to store all of this in the session.

**Review by solvip — CHANGES_REQUESTED** [review]
One nit; one more serious thing.

Multiple concurrent clients will race as-is.

I can attack this if you need help.

File: src/main/java/is/ru/tictactoe/webui/WebUI.java (hunk 2 above)
Comment (sveinng) [reply src/main/java/is/ru/tictactoe/webui/WebUI.java]: I was afraid of this - fear confirmed!

File: src/main/java/is/ru/tictactoe/webui/WebUI.java (hunk 1 above)
Comment (sveinng) [reply src/main/java/is/ru/tictactoe/webui/WebUI.java]: True - leftover method from initial implementation.
All fixed and pretty now.

File: src/main/java/is/ru/tictactoe/webui/WebUI.java (hunk 2 above)
Comment (sveinng) [reply src/main/java/is/ru/tictactoe/webui/WebUI.java]: All variables now live in session.

**Review by solvip — APPROVED** [review]
Awesome - ship it

**Pull request merged**